src/plugins/interactive_setup/server/routes/enroll.ts

```diff
@@ -6,26 +6,143 @@
  * Side Public License, v 1.
  */
 
+import { constants } from 'fs';
+import fs from 'fs/promises';
+import path from 'path';
+
 import { schema } from '@kbn/config-schema';
 
 import type { RouteDefinitionParams } from './';
 
 /**
  * Defines routes to deal with Elasticsearch `enroll_kibana` APIs.
  */
-export function defineEnrollRoutes({ router }: RouteDefinitionParams) {
+export function defineEnrollRoutes({
+  router,
+  logger,
+  initializerContext,
+  core,
+  completeSetup,
+}: RouteDefinitionParams) {
   router.post(
     {
       path: '/internal/interactive_setup/enroll',
       validate: {
-        body: schema.object({ token: schema.string() }),
+        body: schema.oneOf([
+          schema.object({
+            hosts: schema.arrayOf(schema.string(), { minSize: 1 }),
```

Comment:
__question:__ did you have a chance to check that ES client automatically uses second host if the first host doesn't work? I had impression that it only switches between hosts from time to time and, for our use case, we may need to iterate through hosts manually until we find one that's accessible.

Reply:
I'll restrict hosts to `maxSize: 1` to be safe. You can only enter one host using the UI anyways so there's no need to support multiple right now. I wanted to support an array in the API in case we change our minds so that that's not a breaking change but don't think that allowing users to specify multiple hosts is a requirement for the first iteration.